docs/dev.rst

```diff
@@ -335,3 +334,49 @@ the testshell:
    app@xxx:/app$ pytest tests/test_app.py
 
    <pytest output>
+
+
+How to
+======
+
+How to set up a development container for VS Code
+-------------------------------------------------
+The repository contains configuration files to build a
+`development container <https://containers.dev/>`_ in the `.devcontainer`
+directory. If you have the "Dev Containers" extension installed in VS Code, you
+should be prompted whether you want to reopen the folder in a container on
+startup. You can also use the "Dev containers: Reopen in container" command
+from the command palette. The container has all Python requirements installed.
+IntelliSense, type checking, code formatting with Ruff and running the tests
+from the test browser are all set up to work without further configuration.
+
+VS Code should automatically start the container, but it may need to be built on
+first run:
+
+.. code-block:: shell
+
+   $ make devcontainerbuild
+
```

Comment:
Is it worth noting here this one-time command needed for Mac users? https://github.com/mozilla-services/tecken/pull/2872#issuecomment-1881438989 .

I realize this is Mac only and the file system implementation used by Docker Desktop for Mac could change to where this isn't needed anymore.

Reply:
how about this?
```suggestion

Additionally on mac there is the potential that running git from inside any
container that mounts the current directory to `/app`, such as the development
container, will fail with `fatal: detected dubious ownership in repository at
'/app'`. This is likely related to `mozilla-services/tecken#2872
<https://github.com/mozilla-services/tecken/pull/2872>`_, and can be treated by
running the following command, which will probably throw exceptions on some git
read-only objects that are already owned by app:app, so that's fine:

.. code-block:: shell
   $ chown -R app:app /app
```

Reply:
For reST directives, you should have a blank line between options and content.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block

Reply:
The new docs section looks great! To @willkg 's point, yeah `make docs` is throwing for the lack of new line between `.. code-block:: shell` and the code in the code block IIUC. Sorry didn't catch that before.

```diff
diff --git a/docs/dev.rst b/docs/dev.rst
index 9397068..178c03c 100644
--- a/docs/dev.rst
+++ b/docs/dev.rst
@@ -370,6 +370,7 @@ change the default ports exposed by docker compose to avoid conflicts with
 similar services, for example:
 
 .. code-block:: shell
+
    EXPOSE_ELIOT_PORT=8100
    EXPOSE_SENTRY_PORT=8190
    EXPOSE_STATSD_PORT=8181
```